pkg/virt-controller/watch/clone/vm-target.go

```diff
@@ -197,3 +200,13 @@ func generateFirmwareUUIDPatches(firmware *k6tv1.Firmware) (patches []string) {
 
 	return []string{firmwareUUIDPatch}
 }
+
+func generateHostnamePatches(cloneHostname string) (patches []string) {
+	const hostnamePatchPattern = `{"op": "replace", "path": "/spec/template/spec/hostname", "value": "%s"}`
+
+	if cloneHostname == "" {
+		return nil
+	}
+	patch := fmt.Sprintf(hostnamePatchPattern, cloneHostname)
+	return []string{patch}
```

Comment:
Would you mind to use the new `PatchSet` https://github.com/kubevirt/kubevirt/pull/11599 ?